rundeckapp/grails-spa/packages/ui-trellis/src/app/components/readme-motd/EditProjectFile.vue

```diff
@@ -0,0 +1,161 @@
+                <div class="card-header">
+                  <h3 class="card-title">{{  $t("edit.file.project", [this.filename, this.project]) }}</h3>
+                </div>
+                <div class="card-content">
+                  <div class="help-block">
+                    <details class="details-reset more-info">
+                      <summary>
+                        <span v-html="$t('file.readme.help.markdown')"></span>
+                        <span class="more-indicator-verbiage more-info-icon"><i class="glyphicon glyphicon-chevron-right "></i></span>
+                        <span class="less-indicator-verbiage more-info-icon"><i class="glyphicon glyphicon-chevron-down "></i></span>
+                      </summary>
+                      <span v-html="$t('file.readme.help.html')"></span>
+                    </details>
+                  </div>
+                  <ace-editor v-model="fileText"
+                              :soft-wrap-control="true"
+                              height="250"
+                              width="100%"
+                              lang="markdown"
+                              :readOnly="false"
+                  />
+                </div>
+                <div class="card-footer">
+                  <button type="button" class="btn btn-default reset_page_confirm" @click="createProjectHomeLink">Cancel</button>
+                  <button type="submit" class="btn btn-cta reset_page_confirm" @click="saveProjectFile">Save</button>
+                  <template v-if="displayConfig.includes('none')">
+                    <span class="text-warning text-right">
+                      <template v-if="authAdmin">
+                        {{ $t("file.warning.not.displayed.admin.message") }}
+                        <a :href="createProjectConfigureLink"> {{ $t("project.configuration.label") }} </a>
+                      </template>
+                      <template v-else>
+                        {{ $t("file.warning.not.displayed.nonadmin.message") }}
+                      </template>
+                    </span>
+                  </template>
+                </div>
+              </div>
+            </div>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import {defineComponent} from "vue";
+import AceEditorVue from "@/library/components/utils/AceEditorVue.vue";
+import AceEditor from "@/library/components/utils/AceEditor.vue";
+import {getRundeckContext} from "@/library";
+import {url} from  "@/library/rundeckService"
+import {Notification} from "uiv";
+
+const rundeckClient = getRundeckContext().rundeckClient
+export default defineComponent({
+  name: "EditProjectFile",
+  components: {AceEditor, AceEditorVue},
+  data() {
+    return {
+      fileText: '',
+      markdownSectionOpen: false,
+      errorMsg: ''
+    };
+  },
+  props: {
+    filename: '',
+    displayConfig: [],
+    project: '',
+    authAdmin: false,
+  },
```

Comment:
issue: ideally for props we use the object syntax and type them, ie:

```
filename: {
   type: string,
   default: ''
}
```

The benefit of this approach is that you can add validation or mark a prop as required. Could you please change them for this format?